### PR TITLE
feat: support generation of cronjobs

### PR DESCRIPTION
**Pull request by Sgitario**
More information in the feature request: https://github.com/dekorateio/dekorate/issues/855
Fix https://github.com/dekorateio/dekorate/issues/855

### CODE REVIEW

**Review by aureamunoz — COMMENTED** [review]
LGTM but docs are missing, right?

**Comment by Sgitario** [comment]
> LGTM but docs are missing, right?

Ups. Added.

**Review by aureamunoz — APPROVED** [review]
Perfect. thanks

**Review by aureamunoz — CHANGES_REQUESTED** [review]
mmm, I sent the ok too fast. Could you add a link to the [spec](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec) also? And maybe quote the fact that Dekorate should use the pod template image configuration if user doesn't provide a cron specific one?

**Comment by Sgitario** [comment]
> mmm, I sent the ok too fast. Could you add a link to the [spec](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec) also? And maybe quote the fact that Dekorate should use the pod template image configuration if user doesn't provide a cron specific one?

- Where do you think I should the link in? 
- I don't understand your last sentence

File: docs/documentation/kubernetes.md [hunk 1]
```diff
@@ -320,6 +320,31 @@ Currently, the supported annotations for adding jobs are:
...
+  }
+}
+```    
+
+Or via configuration properties at the file `application.properties`:
+
+    dekorate.kubernetes.cron-jobs[0].name=say-hello
+    dekorate.kubernetes.cron-jobs[0].schedule=* * * * *
+    dekorate.kubernetes.cron-jobs[0].containers[0].image=docker.io/user/hello
+
```
Comment (aureamunoz) [inline docs/documentation/kubernetes.md:341]:
```suggestion
Dekorate CronJobs configuration follows the [Kubernetes CronJobs specification](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec). 

If the user doesn't provide CronJob container image, the pod template image configuration will be used.
```

File: docs/documentation/openshift.md [hunk 2]
```diff
@@ -95,3 +95,28 @@ Currently, the supported annotations for adding jobs are:
...
+  }
+}
+```    
+
+Or via configuration properties at the file `application.properties`:
+
+    dekorate.openshift.cron-jobs[0].name=say-hello
+    dekorate.openshift.cron-jobs[0].schedule=* * * * *
+    dekorate.openshift.cron-jobs[0].containers[0].image=docker.io/user/hello
+
```
Comment (aureamunoz) [inline docs/documentation/openshift.md:117]:
```suggestion
Dekorate CronJobs configuration follows the [Kubernetes CronJobs specification](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec). 

If the user doesn't provide CronJob container image, the pod template image configuration will be used.
```

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Review by aureamunoz — APPROVED** [review]

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**